x-pack/metricbeat/module/aws/utils.go

```diff
@@ -0,0 +1,72 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package aws
+
+import (
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/cloudwatchiface"
+	"github.com/pkg/errors"
+
+	s "github.com/elastic/beats/libbeat/common/schema"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// GetStartTimeEndTime function uses durationString to create startTime and endTime for queries.
+func GetStartTimeEndTime(durationString string) (startTime time.Time, endTime time.Time, err error) {
+	endTime = time.Now()
+	duration, err := time.ParseDuration(durationString)
+	if err != nil {
+		return
+	}
+	startTime = endTime.Add(duration)
+	return startTime, endTime, nil
+}
+
+// GetMetricDataPerRegion function uses MetricDataQueries to get metric data output.
+func GetMetricDataPerRegion(metricDataQueries []cloudwatch.MetricDataQuery, nextToken *string, svc cloudwatchiface.CloudWatchAPI, startTime time.Time, endTime time.Time) (*cloudwatch.GetMetricDataOutput, error) {
+	getMetricDataInput := &cloudwatch.GetMetricDataInput{
+		NextToken:         nextToken,
+		StartTime:         &startTime,
+		EndTime:           &endTime,
+		MetricDataQueries: metricDataQueries,
+	}
+
+	reqGetMetricData := svc.GetMetricDataRequest(getMetricDataInput)
+	getMetricDataOutput, err := reqGetMetricData.Send()
+	if err != nil {
+		logp.Error(errors.Wrap(err, "Error GetMetricDataInput"))
+		return nil, err
```

Comment:
It's not critical but try to add context to the error when return. I know it's a pain to have _three liners_ like this
```go
err = errors.Wrap(err, "Error GetMetricDataInput")
logp.Error(err)
return nil, err
```
But we have nothing else until we find something better. What I'd like to avoid is to debug some error that after passing for 20 methods it just returns `Unexpected end of JSON input` because all functions did `return nil, err`. It's like natural language _stack trace_ xD

Reply:
👍 good point! I will try to be consistent add context to the error! Thanks!